### PR TITLE
[MM-51253][MM-51275]

### PR DESCRIPTION
**Pull request by avas27JTG**
Add check for showing user's fullname on subscription details

### CODE REVIEW

File: server/plugin/api.go [hunk 1]
```diff
@@ -339,10 +339,15 @@ func (p *Plugin) handleCreateSubscription(w http.ResponseWriter, r *http.Request
 		return
 	}
 
-	createdByDisplayName := fmt.Sprintf("%s %s", user.FirstName, user.LastName)
-	if len(strings.TrimSpace(createdByDisplayName)) == 0 {
-		createdByDisplayName = user.Username // If user's first/last name doesn't exist then show username as fallback
+	createdByDisplayName := user.Username
+
+	showFullName := p.API.GetConfig().PrivacySettings.ShowFullName
+	// If "PrivacySettings.ShowFullName" is true then show user's first/last name
```
Comment (Kshitij-Katiyar) [inline server/plugin/api.go:345]:
```suggestion
	// If "PrivacySettings.ShowFullName" is true then show the user's first/last name
```

File: server/plugin/api.go [hunk 2]
```diff
@@ -339,10 +339,15 @@ func (p *Plugin) handleCreateSubscription(w http.ResponseWriter, r *http.Request
...
+	// If user's first/last name doesn't exist then show username as fallback
```
Comment (Kshitij-Katiyar) [inline server/plugin/api.go:346]:
```suggestion
	// If the user's first/last name doesn't exist then show the username as fallback
```

File: server/plugin/api_test.go [hunk 3]
```diff
@@ -662,6 +662,11 @@ func TestHandleCreateSubscriptions(t *testing.T) {
 				FirstName: "mockCreatedBy",
 			}, nil)
 
+			privacySettings := model.PrivacySettings{}
```
Comment (Kshitij-Katiyar) [inline server/plugin/api_test.go:665]: maybe something like this ?
```suggestion
			privacySettings := model.PrivacySettings{
			ShowFullName = &showFullName
			}
```

**Review by Kshitij-Katiyar — APPROVED** [review]

**Review by Kshitij-Katiyar — APPROVED** [review]

**Comment by avas27JTG** [comment]
closing this as the same is merged into `master` on the MM repo will sync the internal `master`
https://github.com/mattermost/mattermost-plugin-azure-devops/pull/148

**Pull request closed without merging**